### ISSUE TITLE
Accessibility: Modal dialog lacks proper markup; title; focus trap

### ISSUE THREAD

**Issue by nedredmond** [inklewriter/freeinklewriter#76]
[Modal dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog#accessibility) lacks proper markup and accessible name. Also, I am able to focus and interact with elements outside of the modal dialog, which is in violation of HTML spec.

EDIT: I see that there is an `eventabsorber` overlay, but this does not account for keyboard access / focus.

## Recommended fix:
Refactor to use browser-native modal dialog functionality ([see MDN for explanation and examples](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog#accessibility)). This will come with default accessibility support.

## ADA standard violations:
- https://www.w3.org/WAI/WCAG22/Understanding/focus-order.html
- https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html
- https://www.w3.org/WAI/WCAG21/Understanding/name-role-value
